wgpu/src/backend/web.rs

```diff
@@ -1113,13 +1113,19 @@ impl crate::context::Context for Context {
         _adapter: &Self::AdapterId,
         _adapter_data: &Self::AdapterData,
     ) -> wgt::SurfaceCapabilities {
+        use wgt::TextureFormat::{Bgra8Unorm, Rgba16Float, Rgba8Unorm};
+        let preferred_format = self.0.get_preferred_canvas_format();
+        let mut formats = vec![Rgba8Unorm, Bgra8Unorm, Rgba16Float];
+        if let Some(index) = formats
+            .iter()
+            .position(|format| map_texture_format(*format) == preferred_format)
+        {
+            formats.swap(0, index);
+        }
```

Comment:
If it's not found, it needs to be inserted.

Reply:
`preferred_format` is of type `web_sys::GpuTextureFormat`, and there is no existing fn to map it back to `wgt::TextureFormat`. Do I need to add a fn similar to `map_texture_format` in web.rs to handle this mapping?

Reply:
Ah yeah, I think you do need to add that unfortunately

Reply:
From the [WebGPU Spec](https://www.w3.org/TR/webgpu/#dom-gpu-getpreferredcanvasformat):
> getPreferredCanvasFormat()
Returns an optimal [GPUTextureFormat](https://www.w3.org/TR/webgpu/#enumdef-gputextureformat) for displaying 8-bit depth, standard dynamic range content on this system. Must only return ["rgba8unorm"](https://www.w3.org/TR/webgpu/#dom-gputextureformat-rgba8unorm) or ["bgra8unorm"](https://www.w3.org/TR/webgpu/#dom-gputextureformat-bgra8unorm).

Do we currently have the option to not handle the "not found" case?

Reply:
Oh well, nevermind then, could you just note this requirement in a comment in the code

Reply:
Sure, it has been added.